src/ucp/core/ucp_request.inl

```diff
@@ -512,6 +515,20 @@ ucp_recv_desc_init(ucp_worker_h worker, void *data, size_t length,
     return status;
 }
 
+static UCS_F_ALWAYS_INLINE void
+ucp_recv_desc_release(ucp_recv_desc_t *rdesc)
+{
+    ucs_trace_req("release receive descriptor %p", rdesc);
+    if (ucs_unlikely(rdesc->flags & UCP_RECV_DESC_FLAG_UCT_DESC)) {
+        /* uct desc is slowpath */
+        uct_iface_release_desc(UCS_PTR_BYTE_OFFSET(rdesc,
+                                                   -(UCP_WORKER_HEADROOM_PRIV_SIZE -
+                                                     rdesc->priv_length)));
+    } else {
+        ucs_mpool_put_inline(rdesc);
```

Comment:
what should be priv_length here? do we need an assertion?

Reply:
we do not need it here